applications/trilinos_application/custom_python/trilinos_pointer_wrapper.h

```diff
@@ -0,0 +1,59 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					     Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ruben Zorrilla
+//
+
+#if !defined(KRATOS_TRILINOS_POINTER_WRAPPER_H_INCLUDED)
+#define KRATOS_TRILINOS_POINTER_WRAPPER_H_INCLUDED
+
+// System includes
+
+// External includes
+
+//Trilinos includes
+#include "Epetra_Vector.h"
+#include "Epetra_FEVector.h"
+
+// Project includes
+#include "trilinos_application.h"
+#include "trilinos_space.h"
+
+namespace Kratos
+{
+
+typedef TrilinosSpace<Epetra_FECrsMatrix, Epetra_FEVector> TrilinosSparseSpaceType;
+
+class AuxiliaryMatrixWrapper
+{
+  public:
+    AuxiliaryMatrixWrapper(TrilinosSparseSpaceType::MatrixPointerType p) : mp(p){};
```

Comment:
I believe this should be `typename TrilinosSparseSpaceType::MatrixPointerType` (gcc is more flexible with this than the standard, other compilers will not compile this).

Anyway, since you are using this in quite a few places, I'd suggest a `typedef` or `using` statement.

Reply:
I'm compiling with clang, without any problem. In any case, I'll define a `typedef`

Reply:
Maybe I'm wrong then... Try it and if it is not necessary it should not compile.

Reply:
Done.